lib/imagec/docker.go

```diff
@@ -416,6 +426,22 @@ func decodeManifestSchema1(filename string, options Options, registry string) (i
 	return manifest, digest, nil
 }
 
+// verifyDigest checks the manifest digest against the received payload.
+func verifyManifestDigest(digested reference.Canonical, bytes []byte) error {
+	verifier, err := ddigest.NewDigestVerifier(digested.Digest())
+	if err != nil {
+		return err
+	}
+	if _, err = verifier.Write(bytes); err != nil {
+		return err
+	}
+	if !verifier.Verified() {
+		return fmt.Errorf("image manifest verification failed for digest %s", digested.Digest())
+	}
+
+	return nil
+}
+
 // decodeManifestSchema2() reads a manifest schema 2 and creates a Docker
 // defined Manifest structure and returns the digest of the manifest as a string.
 func decodeManifestSchema2(filename string, options Options) (interface{}, string, error) {
```

Comment:
It's not strictly part of this change, but this feels like it should be part of an `interface` where schema1 and schema2 are implementations.